EssentialsProtect/src/com/earth2me/essentials/protect/EssentialsProtectBlockListener.java

```diff
@@ -90,4 +92,44 @@ public void onPortalLight(PortalCreateEvent event) {
             event.setCancelled(prot.getSettingBool(ProtectConfig.prevent_portal_creation));
         }
     }
+
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onSignDye(PlayerInteractEvent event) {
+        Material clickedBlock = event.getClickedBlock().getBlockData().getMaterial();
+        Action action = event.getAction();
+        Material itemInHandMat = event.getPlayer().getInventory().getItemInMainHand().getType();

```

Comment:
This will error out in 1.8.8, as `getItemInMainHand()` didn't exist until 1.9. If I'm not mistaken, the preferred way to handle this is to get the Essentials `User` and use `User#getItemInHand()`, which accounts for both cases.

Reply:
Fixed on commit 2e5296881436ed9fe5cb967a39095de32613f261

Reply:
What also would have probably worked just fine is to exit this method early by checking the MineCraft version (since this is only intended to run on versions 1.15+). See VersionUtil for some handy methods.